### PR TITLE
Redirect to previous page after login fix #711

### LINKED ISSUE #711: Login prompt on Question comment area does not redirect back after login
Hi, this is a `first-timers-only issue`. This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. If you have contributed before, consider leaving this one for someone new, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!
## The Problem

Visit a research note or question like: https://publiclab.org/questions/Anmar/08-19-2016/question-calibration-issues while not logged in, then click "Log in" to comment in the comment section to login. But then you will be redirected to the dashboard, instead of back to the original post. It'd be great to have a `return_to` param as on other login prompts, which is just adding `?return_to=/path` to the end of the link, which is then read here: https://github.com/publiclab/plots2/blob/master/app/views/user_sessions/new.html.erb#L24
## Solution

There are many areas where similar Login links are present like the notes page, question page and the Answers section in questions page. These are all the relevant areas
- https://github.com/publiclab/plots2/blob/master/app/views/notes/show.html.erb#L109
-  https://github.com/publiclab/plots2/blob/master/app/views/notes/_comments.html.erb#L19
- https://github.com/publiclab/plots2/blob/master/app/views/questions/show.html.erb#L61
- https://github.com/publiclab/plots2/blob/master/app/views/questions/_answers.html.erb#L13
- https://github.com/publiclab/plots2/blob/master/app/views/questions/_answer.html.erb#L58

Everywhere a `?return_to=` parameter is needed beside the login url which should be set to `request .path` for redirecting to the current page.
## Steps to Fix
- [x] claim this issue with a comment
- [ ] set up a repository locally and make sure that all tests pass
- [ ] start a pull request (see contributing to Public Lab software) but mark it as "in progress"
- [ ] fix the issue following the steps above
- [ ] try the manual test listed above
- [ ] make sure all tests pass with rake test
- [ ] alert someone via either the developers list (see below) to have your pull request merged. This may take a few extra steps depending on when you do it, but we'll help you out!

Please email the developers list (see https://publiclab.org/developers) if you have questions!

### PR DESCRIPTION
**Pull request by petersowah**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue
* [ ] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer
* [x] `schema.rb.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

File: app/views/questions/_answer.html.erb [hunk 1]
```diff
@@ -61,18 +61,19 @@
...
       </div>
       <div class="inline" id="question-comment-form">
         <form id="answer-<%= answer.id %>-comment-form" class="form-horizontal" action="/comment/answer_create/<%= answer.id%>" data-remote="true">
-          <%= render partial: "questions/form" %>  
+          <%= render partial: "questions/form" %>
         </form>
       </div>
       <% else %>
-        <p><a href="/login">Log in</a> to comment</p> 
+        <!-- <p><a href="/login">Log in</a> to comment</p>  -->
```
Comment (ananyo2012) [inline app/views/questions/_answer.html.erb]: I hope you don't need this to be commented. You can simply remove this line.

File: app/views/questions/_answers.html.erb [hunk 2]
```diff
@@ -10,8 +10,9 @@
     <% if current_user %>
       <%= render :partial => "comments/form", :locals => { title: "Post an answer", comment: false, placeholder: "Help users by posting an answer to this question" } %>
     <% else %>
-      <p><a href="/signup">Sign up</a> or <a href="/login">Login</a> to post an answer to this question.</p>
-    <% end %> 
-    </div> 
+      <!-- <p><a href="/signup">Sign up</a> or <a href="/login">Login</a> to post an answer to this question.</p> -->
```
Comment (ananyo2012) [inline app/views/questions/_answers.html.erb]: Same here. Just remove this one

File: app/views/questions/_answer.html.erb (hunk 1 above)
Comment (petersowah) [inline app/views/questions/_answer.html.erb]: Ok.. will do shortly. Thanks.

**Comment by petersowah** [comment]
@ananyo2012 changes effected.

File: app/views/questions/_answer.html.erb (hunk 1 above)
Comment (ananyo2012) [inline app/views/questions/_answer.html.erb]: This one is still left

**Comment by ananyo2012** [comment]
@petersowah See [this line](https://github.com/publiclab/plots2/blob/master/app/views/notes/_comments.html.erb#L19). You need to put a `return_to` here as well.

**Comment by petersowah** [comment]
@ananyo2012 I completed the remaining links.

File: app/views/questions/_answer.html.erb (hunk 1 above)
Comment (ananyo2012) [inline app/views/questions/_answer.html.erb]: You are forgetting about this one.

**Comment by ananyo2012** [comment]
@petersowah See the review above. You have missed the comments.

**Comment by ananyo2012** [comment]
@petersowah I see a lot of gem versions changed in Gemfile.lock. Did you ran `bundle update` by any chance? If you did I recommend not doing it from the next time. Reasons - Read this blog post: http://ryanbigg.com/2011/01/why-you-should-run-bundle-update

**Comment by petersowah** [comment]
@ananyo2012 my bad. Thanks for the heads up, I removed the comment and also the Gemfile.lock and run bundle install. thanks for the link to the blog post.

File: .travis.yml [hunk 3]
```diff
@@ -1,5 +1,8 @@
 sudo: required
 
+before_install:
+  - sudo apt-get update
```
Comment (ananyo2012) [inline .travis.yml]: @petersowah Why did you make this change. As I said you shouldn't run `bundle update` like this.

Comment (petersowah) [inline .travis.yml]: @ananyo2012 My tests were failing on Travis CI, I googled the error message and that was the solution. Maybe I shouldn't have made the change then. I didn't run `bundle update`

Comment (petersowah) [inline .travis.yml]: My bad!

Comment (ananyo2012) [inline .travis.yml]: If you see the CI log it says
```
ERROR: Service 'web' failed to build: The command '/bin/sh -c apt-get update -qq && apt-get install -y bundler libmysqlclient-dev ruby-rmagick libfreeimage3 nodejs-legacy npm wget openjdk-7-jre openjdk-7-jdk' returned a non-zero code: 100
```
Actually some packages didn't install properly. In this case I hope just closing and reopening the pull request would do.

Comment (petersowah) [inline .travis.yml]: Alright.

**Comment by ananyo2012** [comment]
@jywarren We are getting more and more build failures these days just due to package installation errors. I thing we should start caching dependencies. @icarito Can you have a look at this?

**Comment by ananyo2012** [comment]
Just remove the previous commit. Also there are also a lot of commits already. Can you try squashing them into one?

**Comment by ananyo2012** [comment]
If you need any help with squashing you can ask in our gitter channel

**Pull request closed without merging**